adding_metadata_sql.md

```diff
@@ -0,0 +1,181 @@
+```
+
+### Step 6: Associate experiments with sites
+
+```sql
+insert into experiments_sites (experiment_id, site_id) values ((select id from experiments where name = 'MAC Season 6: Sorghum BAP'),
+(select id from sites where sitename = 'MAC Field Scanner Season 6 Range 1 Column 1 E'))
+```
+
+When adding a new season for the TERRA REF project, a statement like the following can be used for associating experiments and sites since MAC Field Center sites are consistently named following the format `MAC Field Scanner Season x Range a Column b`.
+
+```sql
+insert into experiments_sites (experiment_id, site_id) 
+   select e.experiment_id, s.site_id 
+        from (select id as experiment_id from experiments where name = 'MAC Season 6: Sorghum BAP')  as e 
+     cross join 
+         (select id as site_id from sites where sitename like 'MAC Field Scanner Season 6%') as s
+```
+
+## Step 7: Associate experiments with treatments
+
+```sql
+insert into experiments_treatments (experiment_id, treatment_id) 
+values ((select id from experiments where name = 'MAC Season 6: Sorghum BAP'),
+(select id from treatments where name = 'MAC Season 6: Sorghum'))
+```
+When adding a new season for the TERRA REF project, a statement like the following can be used to associate experiments with treatments since experiment and treatment names are usually named following the format `MAC Season x: subexperiment name`.
+
+```sql
+insert into experiments_treatments (experiment_id, treatment_id) 
+   select e.experiment_id, t.treatment_id 
+        from (select id as experiment_id from experiments where name like 'MAC Season 6:%')  as e 
+     cross join 
+         (select id as treatment_id from treatments where name like 'MAC Season 6:%') as s
+```
+
+### Step 8: Associate sites with cultivars
+
+```sql
+insert into sites_cultivars (site_id, cultivar_id) 
+values ((select id from sites where sitename = 'MAC Field Scanner Season 8 Range 1 Column 1 E'),
+(select id from cultivars where name = 'Tiburon' and specie_id = 2588))
+```
+
+### Step 9: Associate sites with citations
+
+```sql
+insert into citations_sites (citation_id, site_id)
+values ((select id from citations where author = 'Newcomb, Maria' and year = 2016 and title = 'MAC Field Activities'),
+(select id from sites where sitename = 'MAC Field Scanner Season 6 Range 1 Column 1 E'))
+```
+When adding a new season for the TERRA REF project, a statement like the following can be used to associate citations with sites since MAC Field Center sites are consistently named following the format `MAC Field Scanner Season x Range a Column b`
+
+```sql
+insert into citations_sites (citation_id, site_id) 
+   select c.id, s.id 
+      from (select id from citations where author = 'Newcomb, Maria') 
+     AS c 
+      cross join 
+         (select id from sites where sitename like 'MAC Field Scanner Season 6%') AS s;
+```
+
+
+## Removing duplicate records
+
```

Comment:
If you envision people actually _using_ this function, I could write a much better version which would at least get rid of the easy-to-fix defects.

That said, I think the introduction could be much clearer about describing the use case for the function:

---
"It sometimes happens that multiple rows in a BETYdb table contain essentially the same information.  This makes it harder to obtain consistent and comprehensible search results.  A function such as the one that follows could be used to delete a duplicate row and change all references to it to point to a row that we are retaining.  (Note that there are serious problems associated with the use of this function, as outlined in `[this comment in issue 185](https://github.com/PecanProject/bety/issues/185#issuecomment-530554650){target="_blank"}`.  We present it here only to outline what might be possible along the lines of partially automating the correction of data errors.)

"The function takes three arguments: the name of the table we wish to update (as a string), the id number of the row we wish to remove (the "duplicate"), and the id number of the similar row we wish to retain.  For example, if we have two citation rows having essentially the same information having id numbers 286 and 289, we could remove the first and update references to it to point to the second with the statement
```sql
SELECT update_refs_from_to('citations', 286, 289);
```
---

The difficult issues associated with this function are all associated with the distributed data system we have come up with.  It boils down to this: If you use the function to delete a row, one of a number of things may happen:
1. If the row you deleted is owned by your database but the referring rows are owned by a different database, those references will point to nothing as soon as someone re-imports your database.  And even though you updated your copy of those referring rows using the function, the references you updated will be reverted back to point to nothing if and when you re-import from the database where they originated.
2. If both the deleted row and the referring rows are owned by your database, nothing bad should happen.  The function should update the references to point to a retained row and these updates will propagate to other databases that import yours.
3. If the row you deleted is owned by another database, it will reappear as soon as you re-import that other database.

Probably the "safest" use case for the function is this: You create a species row, say, that duplicates an existing species row on another machine.  (If that other machine is one you import from, assume perhaps that it was created since you last imported from that machine.  Equally likely is that you _have_ imported from that other machine recently but that there are no uniqueness constraints in place that prevent you from creating a row with essentially the same information as the one you imported.)  Then you add a bunch of data that refers to the new species row you created.  Then you realize your new species row is redundant and use the function to delete it and update references to it to point to the row you imported.  As long as no one has imported your new species row to their copy of the database and started adding data that refers to it, we are essentially in case 2 above and the update-references function shouldn't cause any problem.

Reply:
```suggestion
It sometimes happens that multiple rows in a BETYdb table are duplicates, e.g., two species or two sites that were independently added but reference the same entity. A function such as the one that follows could be used to delete a duplicate row and change all references to it to point to a row that we are retaining. (Note that there are serious problems associated with the use of this function, as outlined in [this comment in issue 185](https://github.com/PecanProject/bety/issues/185#issuecomment-530554650){target="_blank"}. We present it here only to outline what might be possible along the lines of partially automating the correction of data errors.)

The function takes three arguments: the name of the table we wish to update (as a string), the id number of the row we wish to remove (the "duplicate"), and the id number of the similar row we wish to retain. For example, if we have two citation rows having essentially the same information having id numbers 286 and 289, we could remove the first and update references to it to point to the second with the statement

```sql
SELECT update_refs_from_to('citations', 286, 289);
```

The issues associated with this function are associated with the distributed data system used to sync data across BETYdb instances in the PEcAn network. If you use the function to delete a row, one of a number of things may happen:

1. If the row you deleted is owned by your database but the referring rows are owned by a different database, those references will point to nothing as soon as someone re-imports your database. And even though you updated your copy of those referring rows using the function, the references you updated will be reverted back to point to nothing if and when you re-import from the database where they originated.
2. If both the deleted row and the referring rows are owned by your database, nothing bad should happen. The function should update the references to point to a retained row and these updates will propagate to other databases that import yours.
3. If the row you deleted is owned by another database, it will reappear as soon as you re-import that other database.
Probably the "safest" use case for the function is this: You create a species row, say, that duplicates an existing species row on another machine. (If that other machine is one you import from, assume perhaps that it was created since you last imported from that machine. Equally likely is that you have imported from that other machine recently but that there are no uniqueness constraints in place that prevent you from creating a row with essentially the same information as the one you imported.) Then you add a bunch of data that refers to the new species row you created. Then you realize your new species row is redundant and use the function to delete it and update references to it to point to the row you imported. As long as no one has imported your new species row to their copy of the database and started adding data that refers to it, we are essentially in case 2 above and the update-references function shouldn't cause any problem.
```